Context/ImplicitContextKeyed.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Context;
+
+/**
+ * Represents a value that can be sored within {@see Context}.
+ * Allows storing themselves without exposing a {@see ContextKey}.
+ */
+interface ImplicitContextKeyed
```

Comment:
This mirrors the Java implementation and allows for `Baggage` to follow the same patterns as `Span`. Java apparently allows defining default method implementations on interfaces, so there is some minor duplication, but could introduce a trait similar to what `ContextValueTrait` was doing if it gets out of hand.

Reply:
Sounds like a plan to me.  Following the JAVA spec is perfectly acceptable here!